src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -620,6 +622,21 @@ protected override void BeginRequestProcessing()
             TimeoutControl.SetTimeout(_keepAliveTicks, TimeoutReason.KeepAlive);
         }
 
+        protected override async Task ProcessRequestAsync<TContext>(IHttpApplication<TContext> application, TContext context)
```

Comment:
I'll add a comment here (if this becomes the final fix). This is the logic we want https://github.com/dotnet/coreclr/blob/f0ba9d3435e260d89c44c73afd14dc12301d72ef/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilderCore.cs#L21-L63. Basically what `ExecutionContext.Run` does